### PR TITLE
Notifications: Fetch notifications from the backend (SQCORE-1137)

### PR DESCRIPTION
**Pull request by makingthematrix**
https://wearezeta.atlassian.net/browse/SQCORE-1137

Three subtasks here:
1. The main one: I rewrote parts of `PushService` as `FCMPushHandler` to contact the backend and fetch notifications. For now, I use them only to update `lastId` - a user "preference" which we use to tell the backend what is the latest notification we received.
2. I wrote two unit basic unit tests and the unit test class for more. `FCMPushHandler` belongs to a module which has no access to `AndroidFreeSpec` and some other utils, so this required a few unusual decisions. For example, I copied `TestUserPreferences` and `TestGlobalPreferences` - two classes we used in tests to mock user and global preferences.
3. To be able to write unit tests I had to correct a very old blunder in the codebase. From the backend, we get a payload that we want to save to the database very quickly. So even though the payload is a json, we don't deserialize it. We just cut it into a sequence of smaller jsons (it's a json array) and save them as strings. But, for some reason, for the last few years, we handled it in the code as `org.json.JSONArray` which is provided by Android and required either the Android framework or Robolectric to work. ([look here](https://stackoverflow.com/questions/22013771/android-java-lang-runtimeexception-stub-exception-while-parsing-string-into-j)). Robolectric causes its own set of problems, so instead I decided to refactor that `JSONArray` to `Vector[EncodedEvent]` where `EncodedEvent` is a newtype over `String`. Nothing changes in the DB schema.

Uff. I hope that from now on writing unit tests will be a breeze.

PS. Please advise me about the naming of classes. "Service" in this context means an Android service so I don't want to use this name as we do in the ZMS module. On the other hand, "handler" sounds like something people use when they have no idea how to name things.

### CODE REVIEW

**Comment by AndroidBob** [comment]
Build Success
ID:1
URL:[Link to Buildjob](https://10.10.124.134/job/wire-android/job/PR-3583/1/)

**Comment by AndroidBob** [comment]
Build Success
ID:2
URL:[Link to Buildjob](https://10.10.124.134/job/wire-android/job/PR-3583/2/)

**Comment by AndroidBob** [comment]
Build Success
ID:3
URL:[Link to Buildjob](https://10.10.124.134/job/wire-android/job/PR-3583/3/)

**Comment by AndroidBob** [comment]
Build Failure
ID:4
URL:[Link to Buildjob](https://10.10.124.134/job/wire-android/job/PR-3583/4/)

**Comment by AndroidBob** [comment]
Build Success
ID:5
URL:[Link to Buildjob](https://10.10.124.134/job/wire-android/job/PR-3583/5/)

**Comment by AndroidBob** [comment]
Build Failure
ID:6
URL:[Link to Buildjob](https://10.10.124.134/job/wire-android/job/PR-3583/6/)

File: zmessaging/src/test/scala/com/waz/service/push/WSPushServiceSpec.scala [hunk 1]
```diff
@@ -142,16 +142,16 @@ import scala.concurrent.duration._
       Thread.sleep(500)
       fakeWebSocketEvents ! SocketEvent.Opened(webSocket)
 
-      val notification = PushNotificationEncoded(id = Uid(), new JSONArray)
+      val notification = PushNotificationEncoded(id = Uid(), Vector.empty)
       import com.waz.utils.Json.syntax._
-      val responseContent = JsonObjectResponse(notification.toJson)
+      //val responseContent = JsonObjectResponse(notification.toJson)
```
Comment (makingthematrix) [inline zmessaging/src/test/scala/com/waz/service/push/WSPushServiceSpec.scala:147]: This test suite is turned off for now. It consists of a few unit tests for `WSPushService` - so, websockets, not push notifications. But I'm thinking if at the end of this feature I could refactor it and turned it on again.
For now, I just comment out a few lines so it's still compiling.

**Comment by AndroidBob** [comment]
Build Success
ID:7
URL:[Link to Buildjob](https://10.10.124.134/job/wire-android/job/PR-3583/7/)

**Review by mdimjasevic — APPROVED** [review]

**Review by ohassine — APPROVED** [review]

**Comment by AndroidBob** [comment]
Build Failure
ID:8
URL:[Link to Buildjob](https://10.10.124.134/job/wire-android/job/PR-3583/8/)

**Pull request merged**